src/main/java/com/github/crimsondawn45/fabricshieldlib/lib/object/FabricShieldItem.java

```diff
@@ -93,43 +64,28 @@ public FabricShieldItem(Settings settings, int coolDownTicks, ToolMaterial mater
      * @param repairItemTag  item tag for repairing shield.
      */
     public FabricShieldItem(Settings settings, int coolDownTicks, int enchantability, TagKey<Item> repairItemTag) {
-        super(settings); //Make durability match material
-
-        //Register dispenser equip behavior
-        DispenserBlock.registerBehavior(this, ArmorItem.DISPENSER_BEHAVIOR);
-
-        //Register that item has a blocking model
-        if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT) {
-            this.RegisterModelPredicate();
-        }
-
-        this.coolDownTicks = coolDownTicks;
-        this.enchantability = enchantability;
-        this.repairType = RepairItemType.TAG;
-        this.repairTag = repairItemTag;
+        this(settings, coolDownTicks, enchantability, Registries.createEntryLookup(Registries.ITEM).getOrThrow(repairItemTag));
     }
 
     /**
      * @param settings       item settings.
      * @param coolDownTicks  ticks shield will be disabled for when it with axe. Vanilla: 100
-     * @param enchantability enchantability of shield. Vanilla: 9
-     * @param repairItemTags list of item tags for repairing shield.
+     * @param enchantability enchantability of shield. Vanilla: 14
+     * @param repairItems    list of items/tags for repairing shield.
      */
-    public FabricShieldItem(Settings settings, int coolDownTicks, int enchantability, Collection<TagKey<Item>> repairItemTags) {
-        super(settings);
+    public FabricShieldItem(Settings settings, int coolDownTicks, int enchantability, @Nullable RegistryEntryList<Item> repairItems) {
+        super((repairItems == null ? settings : settings.component(DataComponentTypes.REPAIRABLE, new RepairableComponent(repairItems))).enchantable(enchantability).equippableUnswappable(EquipmentSlot.OFFHAND));
 
         //Register dispenser equip behavior
-        DispenserBlock.registerBehavior(this, ArmorItem.DISPENSER_BEHAVIOR);
+        // TODO: This is no longer necessary if this item has DataComponentTypes.EQUIPPABLE
```

Comment:
Is there a way we can have that component be a default for all FabricShields?

Reply:
It would be hard if we dont define a base Item class for all FabricShields.
The vanilla is moving away from using the Item class type to distinguish between items, so we should avoid assuming the modded shields are instances of the vanilla shield or the FabricShieldBase.

Reply:
Ah I see, ok. So should we just keep the implementation we have here or should I direct molders to add said component on the wiki guide for shields?